src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,14 +13,15 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
+
+import java.com.google.sps.data.RedditData;
```

Comment:
Java is not included in the package name, so if you remove the prefix `java.`, I think it'll work

Reply:
When I removed it and ran this is what I got. 
![image](https://user-images.githubusercontent.com/44912212/86188379-71b65d80-bb0c-11ea-8ce2-f7d9e2de1680.png)
The errors go on for quite a few more lines all the same but different lines.

Reply:
You need to clean up the RedditData file, there are a few things that don't quite work as is:
  - In Java, you need to have a top level class, which you don't have right now. The easiest way to tackle this is to create a separate file for: RedditData, RedditPost, RedditService. 
 - Then, rename your existing RedditData file into something like `RedditApi` and include the following code:

```java
public class RedditApi {
  @SuppressWarnings("unchecked")
    public void getRedditPosts() throws IOException {
    RedditService service = retrofit.create(RedditService.class);
    Call<RedditData> call = service.getPosts("GoogleFi", "hot", 3);
 
    RedditData redditData = call.execute().body();
    List<Map<String, Object>> children = (List<Map<String, Object>>) redditData.data.get("children");
 
    List<RedditPost> redditPosts = new ArrayList<>();
    for (Map<String, Object> child : children) {
        RedditPost redditPost = new RedditPost();
        redditPost.title = (String) ((Map<String, Object>)child.get("data")).get("title");
        redditPost.content = (String) ((Map<String, Object>) child.get("data")).get("selftext");
        redditPost.url = (String) ((Map<String, Object>) child.get("data")).get("url");
        redditPost.id = (String) ((Map<String, Object>) child.get("data")).get("id");
    }
}
```

If you follow this, you'll have the following file tree:

- com/google/sps/servlets/
   - DataServlet
   - RedditApi
   - RedditData
   - RedditPost
   - RedditService (which is an interface)

From there, you'll have a much cleaner state and other errors will be a bit more readable.

Now, you also need to add Retrofit as a project dependency: https://mvnrepository.com/artifact/com.squareup.retrofit2/parent/2.8.1

```
<dependency>
    <groupId>com.squareup.retrofit2</groupId>
    <artifactId>parent</artifactId>
    <version>2.8.1</version>
    <type>pom</type>
</dependency>

<dependency>
    <groupId>com.squareup.retrofit2</groupId>
    <artifactId>converter-gson</artifactId>
    <version>2.9.0</version>
</dependency>
```